src/File.php

```diff
@@ -485,6 +489,65 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+
+    /**
+     * Check if the File has a CanViewType set or inherited at "LoggedInUsers or above"
+     *
+     * @return bool
+     */
+    public function hasRestrictedAccess(): bool
+    {
+        return $this->hasRestrictedPermissions($this);
+    }
+
+    /**
+     * Recursively determine whether a File has, or inherits, restricted permissions.
+     *
+     * This custom permission check is essentially a couldView() if the record was published for an
+     * anonymous (not-logged-in) user
+     *
+     * Ideally we wouldn't need this custom logic and instead use File->canView() which utilises
+     * InheritedPermission::batchPermissionCheck() and we'd wrap this in `Member::actAs(null,` and
+     * `Versioned::set_reading_mode('Stage.Stage')`, however the anonymous user will fail the
+     * `Versioned::canViewVersioned()` check
+     *
+     * @param File $file
+     * @return bool
+     */
+    private function hasRestrictedPermissions(File $file): bool
+    {
+        $id = $this->ID;
```

Comment:
Believe it or not the function still returns the correct result if you incorrectly have `$this->ID` instead of the correct `$file->ID`, though the caching no longer works

Have updated